File: js/연결 요소의 개수(11724).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, M] = input[0].split(' ').map(Number);
const graph = Array.from({length: N+1}, () => []);
for (let i=0; i < M; i++) {
    const [a, b] = input[i+1].split(' ').map(Number);
    graph[a].push(b);
    graph[b].push(a);
}

const visited = Array.from({length: N+1}, () => false);

function dfs(v) {
    visited[v] = true;
    for (let i of graph[v]) {
        if (visited[i] === false) {
            visited[i] = true;
            dfs(i);
        }
    }
}

let cnt = 0;

for (let i=1; i <= N; i++) {
    if (visited[i] === false) {
        cnt++;
        dfs(i);
    } 
}

console.log(cnt);

File: python/연결 요소의 개수(11724).py
import sys

sys.setrecursionlimit(10**7)
input = sys.stdin.readline

N, M = map(int, input().split())
graph = [[] for _ in range(N+1)]

for i in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (N+1)

def dfs(v):
    visited[v] = True
    for i in graph[v]:
        if visited[i] == False:
            visited[i] = True
            dfs(i)

cnt = 0
for i in range(1, N+1):
    if visited[i] == False:
        cnt += 1
        dfs(i)

print(cnt)